src/utils/trust/trust.involved.ts

```diff
@@ -0,0 +1,208 @@
+  },
+  pageData: {
+    beneficialOwnerTypeTitle: Record<string, string>;
+    trusteeTypeTitle: Record<string, string>;
+    trusteeType: typeof TrusteeType;
+    individualTrusteeData: IndividualTrustee[];
+    formerTrusteeData: TrustHistoricalBeneficialOwner[];
+    checkYourAnswersUrl: string;
+    beneficialOwnerUrlDetach: string;
+    trustData: CommonTrustData,
+  } & TrustWhoIsInvolved,
+  formData?: TrustWhoIsInvolvedForm,
+  errors?: FormattedValidationErrors,
+  url: string,
+};
+
+const getPageProperties = (
+  req: Request,
+  isUpdate: boolean,
+  formData?: TrustWhoIsInvolvedForm,
+  errors?: FormattedValidationErrors,
+): TrustInvolvedPageProperties => {
+  const trustId = req.params[config.ROUTE_PARAM_TRUST_ID];
+
+  const appData = getApplicationData(req.session);
+
+  return {
+    backLinkUrl: getBackLink(isUpdate, trustId),
+    templateName: getTemplateName(isUpdate),
+    pageParams: {
+      title: TRUST_INVOLVED_TEXTS.title,
+    },
+    pageData: {
+      trustData: mapCommonTrustDataToPage(appData, trustId),
+      ...mapTrustWhoIsInvolvedToPage(appData, trustId),
+      beneficialOwnerTypeTitle: TRUST_INVOLVED_TEXTS.boTypeTitle,
+      trusteeTypeTitle: TRUST_INVOLVED_TEXTS.trusteeTypeTitle,
+      individualTrusteeData: getIndividualTrusteesFromTrust(appData, trustId),
+      formerTrusteeData: getFormerTrusteesFromTrust(appData, trustId),
+      trusteeType: TrusteeType,
+      checkYourAnswersUrl: getCheckYourAnswersUrl(isUpdate),
+      beneficialOwnerUrlDetach: `${config.TRUST_ENTRY_URL}/${trustId}${config.TRUST_BENEFICIAL_OWNER_DETACH_URL}`,
+    },
+    formData,
+    errors,
+    url: getServiceUrl(isUpdate),
+  };
+};
+
+const getTrustInvolvedPage = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  isUpdate: boolean
+): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const pageProps = getPageProperties(req, isUpdate);
+
+    return res.render(pageProps.templateName, pageProps);
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const postTrustInvolvedPage = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  isUpdate: boolean
+) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    if (req.body.noMoreToAdd) {
+      return safeRedirect(res, getPostTemplateName(isUpdate));
+    }
+    //  check on errors
+    const errorList = validationResult(req);
+
+    if (!errorList.isEmpty()) {
+      const pageProps = getPageProperties(
+        req,
+        isUpdate,
+        req.body,
+        formatValidationError(errorList.array()),
+      );
+
+      return res.render(pageProps.templateName, pageProps);
+    }
+
+    const typeOfTrustee = req.body.typeOfTrustee;
+
+    // the req.params['id'] is already validated in the has.trust.middleware but sonar can not recognise this.
+    let url = `${config.TRUST_ENTRY_URL}/${req.params[config.ROUTE_PARAM_TRUST_ID]}`;
```

Comment:
need to set a different URL here for Update journey

Reply:
actually ignore this, seeing that forward navigation after choosing radio buttons is out of scope for this ticket